core/src/main/java/com/segment/analytics/kotlin/core/platform/EventPipeline.kt

```diff
@@ -64,20 +64,30 @@ internal class EventPipeline(
     }
 
     fun start() {
+        if (running) return
         running = true
+
+        // avoid to re-establish a channel if the pipeline just gets created
+        if (writeChannel.isClosedForSend || writeChannel.isClosedForReceive) {
+            writeChannel = Channel(UNLIMITED)
+            uploadChannel = Channel(UNLIMITED)
+        }
```

Comment:
yeah. that's possible. but we're using `channel.cancel` in `stop`, buffered events are going to be cleared anyway

the only workaround is not to close/cancel channel at all, but that increases the complexity, since we would have to distinguish pause vs stop. if stop, we would want to close anything ongoing completely, for example on application shutdown.